File: TurathAI-Backend-main/src/main/java/pi/turathai/turathaibackend/Entites/Forum.java
package pi.turathai.turathaibackend.Entites;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Forum {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String description;
    private String image;

    private LocalDateTime createdAt;

    @ManyToOne
    @JoinColumn(name = "idUser")
    private User user;

    @OneToMany(mappedBy = "forum", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Comment> comments;
}

File: src/app/core/Models/forum.ts
import { User } from './user';

import { ForumComment } from './forumComment';

export interface Forum {
  id?: number;
  title: string;
  description: string;
  image?: string;
  createdAt?: Date;
  user?: User;
  showComments?: boolean;
  newComment?: string;
  comments?: ForumComment[];
}
export type Comment = ForumComment;


